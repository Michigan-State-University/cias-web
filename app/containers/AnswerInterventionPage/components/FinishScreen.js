import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const FinishScreen = ({ selectAnswer }) => {
  useEffect(() => selectAnswer(), []);
  return <></>;
};

FinishScreen.propTypes = {
  selectAnswer: PropTypes.func,
};

export default FinishScreen;
