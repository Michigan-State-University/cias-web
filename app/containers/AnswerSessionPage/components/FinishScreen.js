import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import FinishScreenLayout from '../layouts/FinishScreenLayout';

const FinishScreen = ({ selectAnswer, formatMessage }) => {
  useEffect(() => selectAnswer(), []);
  return <FinishScreenLayout formatMessage={formatMessage} />;
};

FinishScreen.propTypes = {
  selectAnswer: PropTypes.func,
  formatMessage: PropTypes.func,
};

export default FinishScreen;
