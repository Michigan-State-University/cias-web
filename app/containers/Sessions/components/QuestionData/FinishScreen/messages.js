import { defineMessages } from 'react-intl';

export const scope = 'app.containers.FinishScreen';

export default defineMessages({
  dashboard: {
    id: `${scope}.dashboard`,
    defaultMessage: 'Go to dashboard',
  },
  tooltip: {
    id: `${scope}.tooltip`,
    defaultMessage: 'Visible only for logged in participants',
  },
  closeMySession: {
    id: `${scope}.closeMySession`,
    defaultMessage: 'Close my session',
  },
});
