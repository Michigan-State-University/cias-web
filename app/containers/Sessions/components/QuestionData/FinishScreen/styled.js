import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const StyledLink = styled(Link)`
  font-weight: bold;
  width: auto;
`;
