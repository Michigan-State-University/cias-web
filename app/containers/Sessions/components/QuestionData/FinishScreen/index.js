import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import Column from 'components/Column';
import Row from 'components/Row';

import Button from 'components/Button';
import Tooltip from 'components/Tooltip';

import messages from './messages';

const FinishScreen = ({ intl: { formatMessage } }) => (
  <Column mt={10}>
    <Row mt={50} justify="center" width="100%">
      <Tooltip
        id="tooltip-finish-screen"
        content={formatMessage(messages.tooltip)}
      >
        <Button disabled px={20} width="100%">
          {formatMessage(messages.dashboard)}
        </Button>
      </Tooltip>
    </Row>
  </Column>
);

FinishScreen.propTypes = {
  intl: PropTypes.object.isRequired,
};

export default injectIntl(FinishScreen);
