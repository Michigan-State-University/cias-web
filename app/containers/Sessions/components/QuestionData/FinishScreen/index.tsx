import React from 'react';
import { useIntl } from 'react-intl';

import { InterventionSharedTo } from 'models/Intervention';

import Column from 'components/Column';
import Row from 'components/Row';
import Button from 'components/Button';

import messages from './messages';

import { CommonQuestionProps } from '../types';

type Props = CommonQuestionProps;

const FinishScreen = ({ sharedTo }: Props) => {
  const { formatMessage } = useIntl();
  return (
    <Column mt={10}>
      <Row mt={50} justify="center" width="100%">
        <Button disabled px={20} width="auto">
          {formatMessage(
            messages[
              sharedTo !== InterventionSharedTo.ANYONE
                ? 'dashboard'
                : 'closeMySession'
            ],
          )}
        </Button>
      </Row>
    </Column>
  );
};

export default FinishScreen;
