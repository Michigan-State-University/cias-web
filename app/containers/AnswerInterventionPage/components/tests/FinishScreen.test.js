/**
 *
 * Tests for FinishScreen
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import FinishScreen from '../FinishScreen';

describe('<FinishScreen />', () => {
  const defaultProps = {
    selectAnswer: jest.fn(),
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<FinishScreen {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(<FinishScreen {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
