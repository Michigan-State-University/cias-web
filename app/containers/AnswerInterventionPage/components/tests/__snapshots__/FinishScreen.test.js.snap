// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinishScreen /> Should render and match the snapshot 1`] = `<div />`;
